File: server/src/main/java/com/example/schoolsite/services/impl/AskQuestionServiceImpl.java
package com.example.schoolsite.services.impl;

import com.example.schoolsite.entity.Question;
import com.example.schoolsite.services.AskQuestionService;
import com.example.schoolsite.workWithDatabase.repo.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AskQuestionServiceImpl implements AskQuestionService {

    private final QuestionRepository questionRepository;

    public Question createQuestion(Question question) {
        System.out.println(question);
        Question questionFromRepo = questionRepository.findByQuestion(question.getQuestion());
        if (questionFromRepo == null) {
            questionRepository.save(question);
        }
        return question;
    }
}

File: server/src/main/java/com/example/schoolsite/services/impl/AskQuestionServiceImpl.kt
package com.example.schoolsite.services.impl

import com.example.schoolsite.entity.Question

@Service
@RequiredArgsConstructor
class AskQuestionServiceImpl : AskQuestionService {
    private val questionRepository: QuestionRepository? = null
    fun createQuestion(question: Question): Question {
        val questionFromRepo: Question = questionRepository.findByQuestion(question.getQuestion())
        if (questionFromRepo == null) {
            questionRepository.save(question)
        }
        return question
    }
}